### DISCUSSION TITLE
Show pywal changes without having to restart qtile every time after changing wallpaper

### DISCUSSION THREAD

**Discussion by hamensman** [qtile/qtile#2858] [Q&A] [answered]
I have this pywal code (https://github.com/dylanaraps/pywal/wiki/Customization#qtile) written into my config and map this (https://github.com/LukeSmithxyz/voidrice/blob/master/.local/bin/setbg) script to a keybind to change my wallpapers.

However, after I change my wallpaper, I have to then restart qtile to see the changes (on, say, the bar). And I don't want to do that every time.

Would I need to write a function which runs the script then restarts qtile? Or can it be done without having to restart qtile at all?

**Comment by m-col** [comment, marked as answer]
Currently your best bet would be to restart qtile at the end of the script. In the not-so-distant future there will be a function to reload the config without doing a full restart, and this will apply these changes and be a much smoother experience. See https://github.com/qtile/qtile/pull/2360 for details, or the code if you want to use it now.

**Reply by elParaguayo** [reply]
Worth adding that putting `lazy.reload()` (as per the pywal code linked above) in a function isn't going to do anything as that doesn't actually restart qtile. `lazy` objects are meant to be attached to keybindings so that they're executed at that point.

**Reply by hamensman** [reply]
> In the not-so-distant future there will be a function to reload the config without doing a full restart, and this will apply these changes and be a much smoother experience.

I very much look forward to that!

**Reply by hamensman** [reply]
> Worth adding that putting `lazy.reload()` (as per the pywal code linked above) in a function isn't going to do anything as that doesn't actually restart qtile. `lazy` objects are meant to be attached to keybindings so that they're executed at that point.

So that explains why adding `qtile.lazy.restart()` to the end of a function I tried writing didn't do anything...

**Reply by hamensman** [reply]
I found this command which restarts qtile, and added it to the wallpaper script: `qtile cmd-obj -o cmd -f restart`.

While it more or less does what I want, it's not the smoothest experience. So having a function to reload the config without a full restart would definitely be the most ideal!
